clap_complete/src/shells/bash.rs

```diff
@@ -59,7 +59,7 @@ impl Generator for Bash {
 complete -F _{name} -o bashdefault -o default {name}
 ",
                 name = bin_name,
-                cmd = bin_name.replace("-", "__"),
+                cmd = bin_name.replace('-', "__"),
```

Comment:
> I'm tightening the scope of this PR, and just focusing on warnings produced by the current clippy configuration.

I'm assuming you mean these are `warn` by default in the latest clippy?  If not, we should make sure they are `#![warn(...)]`.

`make clippy-default` came back clean for me with 1.57

Reply:
i didn't add any flags, so i guess these are warn by default in nightly clippy. Happy to add a flag if you wish!

Reply:
If they are available on stable, let's enable them.

Even if they are on a path towards being enabled on stable, we have to wait for it to go through beta, be released, and then two more releases before it becomes our MSRV.

Reply:
done